src/createMap.ts

```diff
@@ -1,74 +1,84 @@
-import React, { useEffect, useRef, useState } from 'react';
 import mapboxgl from 'mapbox-gl';
-import 'mapbox-gl/dist/mapbox-gl.css';
-import Cat from '../Interfaces/Cat';
-import Coordinates from '../Types/Coordinates';
+import Cat from './Interfaces/Cat';
 
-export default function Map({ cats }: { cats: Cat[] }) {
-	const mapContainer = useRef<HTMLDivElement | null>(null); //  Lets you create a reference to a DOM element that stays between renders without triggering a re-render
-	const map = useRef<mapboxgl.Map | null>(null);
+export default function map(
+	mapContainer: React.RefObject<HTMLDivElement | null>,
+	home: [number, number],
+	cats: Cat[],
+	map: React.MutableRefObject<mapboxgl.Map | null>
+) {
+	const accessToken = process.env.REACT_APP_MAPBOX_ACCESS_TOKEN; // My mapbox access token
 
-	const home: Coordinates = [-1.45054, 53.80619];
+	if (!accessToken)
+		throw new Error(
+			'Mapbox access token is not defined in the environment variables'
+		);
 
-	useEffect(() => {
-		const accessToken = process.env.REACT_APP_MAPBOX_ACCESS_TOKEN; // My mapbox access token
+	mapboxgl.accessToken = accessToken;
 
-		if (!accessToken) {
-			throw new Error(
-				'Mapbox access token is not defined in the environment variables'
-			);
-		}
+	if (mapContainer.current) {
+		// Check if mapContainer is not null
+		map.current = new mapboxgl.Map({
+			// Creates new map
+			container: mapContainer.current, // Adds map to map container
+			center: home, // Sets map center to home location
+			zoom: 15,
+		});
+	}
 
-		mapboxgl.accessToken = accessToken;
-		if (mapContainer.current) {
-			// Check if mapContainer is not null
-			map.current = new mapboxgl.Map({
-				// Creates new map
-				container: mapContainer.current, // Adds map to map container
-				center: home, // Sets map center to home location
-				zoom: 15,
-			});
-		}
+	if (map.current) {
+		cats.forEach((cat) => {
+			console.log(cat.history);
 
-		if (map.current) {
-			cats.forEach((cat) => {
+			const historyFormatted = cat.history.map((coord) => [
+				coord.lon,
+				coord.lat,
+			]);
+
+			if (historyFormatted.length > 0) {
 				const markerElement = document.createElement('div'); // Creates cat icon
 				markerElement.style.width = '50px';
 				markerElement.style.height = '50px';
 				markerElement.style.backgroundImage = `url(${cat.image})`;
 				markerElement.style.backgroundSize = 'cover'; // Sets icon ontop of map properly
 
+				console.log('Marker element styles:', markerElement.style);
+
 				const popup = new mapboxgl.Popup({ offset: 25 }) // Creates popup for marker
 					.setHTML(`<h3>${cat.name}</h3>`); // Popup with cat's name
```

Comment:
Probably risky to assume this will always be a third-level heading wherever it's used. Better to not use an h tag here.